### PR TITLE
Switch to using TaggedText (from SymbolDisplayPart) for features that want to display classified text.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@paulvanbrenk @jramsay This will affect you.  Effectively the SignatureHelp types moved ot using TaggedText instead of SymbolDisplayPart. 
i.e. in this code:

```c#
        private static SignatureHelpItem ConvertSignatureHelpItem(JToken arg)
        {
            return new SignatureHelpItem(
                arg.Value<bool>("isVariadic"),
                c => ConvertSymbolDisplayParts(arg["documentation"]),
                ConvertSymbolDisplayParts(arg["prefixDisplayParts"]),
                ConvertSymbolDisplayParts(arg["separatorDisplayParts"]),
                ConvertSymbolDisplayParts(arg["suffixDisplayParts"]),
                arg["parameters"].Select(ConvertSignatureHelpParameter).ToList(),
                Array.Empty<SymbolDisplayPart>());
        }
```

You'll want to use ConvertTaggedTextParts instead.

**Comment by CyrusNajmabadi** [comment]
Closed side is here: https://github.com/dotnet/roslyn-internal/pull/1184

**Comment by CyrusNajmabadi** [comment]
tagging @dotnet/roslyn-ide This helps with streaming-FAR as well.  That's because we use INavigableItems for FAR and they expose TaggedText.  We need to convert that to WPF inlines and htis PR makes that possible.

File: src/Features/Core/Portable/SignatureHelp/SignatureHelpItem.cs [hunk 1]
```diff
@@ -20,29 +20,30 @@ internal class SignatureHelpItem
...
+        public ImmutableArray<TaggedText> DescriptionParts { get; internal set; }
 
-        public Func<CancellationToken, IEnumerable<SymbolDisplayPart>> DocumentationFactory { get; }
+        public Func<CancellationToken, IEnumerable<TaggedText>> DocumentationFactory { get; }
 
-        private static readonly Func<CancellationToken, IEnumerable<SymbolDisplayPart>> s_emptyDocumentationFactory = _ => SpecializedCollections.EmptyEnumerable<SymbolDisplayPart>();
+        private static readonly Func<CancellationToken, IEnumerable<TaggedText>> s_emptyDocumentationFactory =
+            _ => SpecializedCollections.EmptyEnumerable<TaggedText>();
 
         public SignatureHelpItem(
```
Comment (paulvanbrenk) [inline src/Features/Core/Portable/SignatureHelp/SignatureHelpItem.cs]: Can you keep both constructors to make the migration for TS a little less jarring...

Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/SignatureHelp/SignatureHelpItem.cs]: I can try!

**Comment by Pilchie** [comment]
:+1:

**Pull request merged**